tests/test_meshtastic_utils_edge_cases.py

```
@@ -496,7 +539,21 @@ def test_on_meshtastic_message_matrix_relay_failure(self):
+                """
+                Close the given coroutine (if one) and unconditionally raise an exception indicating matrix relay failure.
+
+                Parameters:
+                    coro: A coroutine or any object; if `coro` is a coroutine it will be closed before raising.
+
+                Raises:
+                    Exception: Always raised with the message "Matrix relay failed".
+                """
+                if asyncio.iscoroutine(coro):
+                    coro.close()
+                raise Exception("Matrix relay failed")
```

Ruff (0.15.10)

[warning] Create your own exception

(TRY002)